src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -2131,6 +2134,65 @@ private void checkInternetPermission() {
         }
     }
 
+    protected String serialize(String uniqueUserId) throws UsageAuthenticationException {
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if ((new BrokerProxy(mContext)).canSwitchToBroker()) {
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
+        } else {
+            /* the current serialize/deserialize feature is for first party 
```

Comment:
Please use the phrase "only supports MS apps" rather than referring to first parties.